src/main/java/co/rsk/bitcoinj/script/RedeemScriptParserFactory.java

```diff
@@ -10,27 +10,27 @@ public class RedeemScriptParserFactory {
     private static final byte[] NON_STANDARD_ERP_TESTNET_REDEEM_SCRIPT_SERIALIZED = Utils.HEX.decode("6453210208f40073a9e43b3e9103acec79767a6de9b0409749884e989960fee578012fce210225e892391625854128c5c4ea4340de0c2a70570f33db53426fc9c746597a03f42102afc230c2d355b1a577682b07bc2646041b5d0177af0f98395a46018da699b6da210344a3c38cd59afcba3edcebe143e025574594b001700dec41e59409bdbd0f2a0921039a060badbeb24bee49eb2063f616c0f0f0765d4ca646b20a88ce828f259fcdb955670300cd50b27552210216c23b2ea8e4f11c3f9e22711addb1d16a93964796913830856b568cc3ea21d3210275562901dd8faae20de0a4166362a4f82188db77dbed4ca887422ea1ec185f1421034db69f2112f4fb1bb6141bf6e2bd6631f0484d0bd95b16767902c9fe219d4a6f5368ae");
     private static final Logger logger = LoggerFactory.getLogger(RedeemScriptParserFactory.class);
 
+
     private RedeemScriptParserFactory() { }
 
-    public static RedeemScriptParser get(List<ScriptChunk> chunks) {
+    public static RedeemScriptParser get(List<ScriptChunk> redeemScriptChunks) {
         // Due to a validation error, during the time this federation existed in testnet
         // bitcoinj-thin would not detect it correctly as an ERP fed
         // We need to keep this behaviour for the given redeem script to keep the consensus in testnet
         List<ScriptChunk> nonStandardErpTestnetRedeemScriptChunks = ScriptParser.parseScriptProgram(
             NON_STANDARD_ERP_TESTNET_REDEEM_SCRIPT_SERIALIZED);
-        if (nonStandardErpTestnetRedeemScriptChunks.equals(chunks)) {
+        if (nonStandardErpTestnetRedeemScriptChunks.equals(redeemScriptChunks)) {
             logger.debug("[get] Received redeem script matches the testnet federation hardcoded one. Return NonStandardErpRedeemScriptParserHardcoded");
             return new NonStandardErpRedeemScriptParserHardcoded();
         }
 
-        if (chunks.size() < 4) {
-            // A multisig redeem script must have at least 4 chunks (OP_N [PUB1 ...] OP_N CHECK_MULTISIG)
-            logger.trace("[get] Less than 4 chunks, return NonStandardErpRedeemScriptParserHardcoded");
-            return new NonStandardErpRedeemScriptParserHardcoded();
+        if (redeemScriptChunks.size() < 4) {
```

Comment:
Do we still need this check? If a script with less than 4 chunks is passed, it shouldn't be able to parse it and throw a ScriptException at the end right?

Reply:
We can let this check to know in advance when a malformed script has been passed.

Reply:
Not so sure about this, having at least 4 chunks doesn't guarantee anything. Looks to me like just one of  various checks that could be made

Reply:
For a redeem script be a multisg, it needs at least four chunks. So far, we are always expecting redeem script to be multi-sig alike. You can find this basic check also in [bitcoinj](https://github.com/rsksmart/bitcoinj/blob/90b4b54b4364d8b8bfc43334df7e6be215370382/core/src/main/java/org/bitcoinj/script/ScriptPattern.java#L205).